cli/src/semgrep/core_output.py

```diff
@@ -191,15 +190,17 @@ def convert_to_rule_match(match: out.CoreMatch) -> RuleMatch:
             fix=fix,
         )
 
-    # TODO: Dict[out.RuleId, RuleMatchSet]
-    findings: Dict[Rule, RuleMatchSet] = {rule: RuleMatchSet(rule) for rule in rules}
-    seen_cli_unique_keys: Set[Tuple] = set()
+    by_unique_key: Dict[Tuple, RuleMatch] = {}
```

Comment:
Is it easy to specify the types in the Tuple?

Reply:
Not really, without it being very verbose. I could maybe define a type alias for the return type of `cli_unique_key` and then use that, although I don't really think it should be transparently a tuple.